cmake/SpectreAddCatchTests.cmake

```diff
@@ -42,11 +42,15 @@ add_custom_target(unit-tests)
 spectre_define_test_timeout_factor_option(UNIT "unit")
 
 # Environment variables for test
-set(_TEST_ENV_VARS "")
+set(_CATCH_TEST_ENV_VARS "")
 # - Disable ASAN's leak sanitizer because Charm++ has false positives
-list(APPEND _TEST_ENV_VARS "ASAN_OPTIONS=detect_leaks=0")
+list(APPEND _CATCH_TEST_ENV_VARS "ASAN_OPTIONS=detect_leaks=0")
 # - Set PYTHONPATH to find Python modules
-list(APPEND _TEST_ENV_VARS "PYTHONPATH=${PYTHONPATH}")
+list(APPEND _CATCH_TEST_ENV_VARS "PYTHONPATH=${PYTHONPATH}")
+# - Disable OpenMP multithreading in ctest because it runs multiple tests in
+#   parallel. Still allows to set this environment variable in the shell when
+#   running test executables manually to benchmark multithreaded performance.
+list(APPEND _CATCH_TEST_ENV_VARS "OMP_NUM_THREADS=1")
```

Comment:
I don't like this solution, but I haven't thought of a better one so far.  This means that running the test executables directly isn't going to do what they are intended to do or what people expect, and there's not going to be an easy way for people to figure out what's going on.

I've been confused in similar ways in the past by tests that do benchmarking, but at least for those the benchmarking code is obvious in the test.

Reply:
The alternative is to disable multithreading in the test main function, which is very easy to do (I can push a quick commit tomorrow). I didn't do that because I wanted to control the benchmark with the environment variable, but it's certainly more predictable. I can add a line to the benchmark to set the num threads at compile time.

Reply:
pushed